IntroComponentWithSign-upForm/script.js

```diff
@@ -0,0 +1,67 @@
+const form  = document.getElementById("form")
+const firstName  = document.getElementById("firstName")
+const lastName  = document.getElementById("lastName")
+const email  = document.getElementById("email")
+const password  = document.getElementById("password")
+//show error message
+const showErrorMessage = ( input , message ) => {
+    const span = document.querySelector("."+input.id)  
+    console.log(span.innerText = message);
+}
+// Show input success message
+const showSuccessMessage = ( input ) => {
+    const span = document.querySelector("."+input.id)
+    span.innerText = "";
+}
+//get field name
+const getFieldName = (input) =>{
+    return input.name;   
+} 
+// check required fields
+const checkRequired = (inputArr) => {    
+    inputArr.forEach( (input)=>{ 
+        if ( input.value.trim() === "")
+        { 
+            showErrorMessage(input , getFieldName(input) +"  " + "cannot be empty")
+             //icon error    
+             var erIcon = document.querySelectorAll(".iconError")
+             for (var i = 0; i < erIcon.length; i++) {
+                erIcon[i].style.visibility = "visible";;
+             }    
```

Comment:
Give the error icon a better/descriptive name! Always a good practice, so anyone else working on this codebase can know right away what this is!

Also, even though declaring a variable with `var`, using `const` and `let` are a bit more descriptive of what are your intentions for these values in the future. For example, a `const` is something that won't ever change (hence the name CONSTANT) and a `let` is a variable that will change at some point in the code!

And one last fix, you had an extra `;` at the end of the line `29`! Another thing that Prettier would catch and fix for you! 😉 
This applies for the consistency of the code (on line `27` you were missing a semicolon).

```suggestion
             const errorIcon = document.querySelectorAll(".iconError");

             for (var i = 0; i < errorIcon.length; i++) {
                errorIcon[i].style.visibility = "visible";
             }    
```